src/python/pants/option/scope.py

```diff
@@ -34,10 +36,21 @@ class ScopeInfo:
     removal_version: Optional[str] = None
     removal_hint: Optional[str] = None
 
-    @property
+    # TODO: We only memoize this to avoid repeating the deprecation warning. Revert back once the
+    #  deprecation is finished.
+    @memoized_property
     def description(self) -> str:
```

Comment:
I couldn't figure out where I could stick a deprecation warning for subsystems. The constructor for `Subsystem` did not work. There wasn't any obvious place that gets called whenever a subsystem is used, but isn't going to flood with repeated warnings.

This will only trigger when running `./pants help`, but not things like `./pants lint`.